cmd/generate/config/rules/algolia.go

```diff
@@ -0,0 +1,23 @@
+package rules
+
+import (
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/secrets"
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func AlgoliaApiKey() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Algolia API Key",
+		RuleID:      "algolia-api-key",
+		Regex:       generateUniqueTokenRegex(`[a-z0-9]{32}`),
```

Comment:
@SimonGurney this should use [generateSemiGenericRegex](https://github.com/zricethezav/gitleaks/blob/master/cmd/generate/config/rules/rule.go#L31-L42) since the algolia token does not have any unique prefix/suffix that is part of the rule. This will slightly change the generated regular expression.

Reply:
hmmm, I thought the opposite was true...  It has no unique suffix/prefix so would use the unique token regex rather than generateSemiGenericRegex which wants unique identifiers?

I must have misread it.

I can't compile the go to test, I'll have to leave it you

Reply:
> It has no unique suffix/prefix so would use the unique token regex

Using the `generateUniqueTokenRegex` means the regular expression can be less strict as there are uniquely identifiable patterns within the token. Using `generateSemiGenericRegex` means that the token follows a generic api token format like a 32 alphanumeric character sequence in this case. 

> I must have misread it.

No worries, this functionality isn't really documented anywhere and is a work in progress. The config generation `cmd` was really just a forcing function for me to finally get the default gitleaks rules in order.

> I can't compile the go to test, I'll have to leave it you

I can merge this as is then update the code so you get contribution credit if you'd like

Reply:
Sounds like a plan 👍